Sources/XiEditor/Annotation.swift

```diff
@@ -0,0 +1,125 @@
+// Copyright 2018 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// Supported annotation types.
+enum AnnotationType: String, CaseIterable {
```

Comment:
Travis CI doesn't seem to like this :(

Reply:
`CaseIterable` is Xcode 10 feature. We're currently at 9.2 👼

Reply:
Okay, if there's a compelling reason to update our swift version that's okay with me.

Reply:
I could also do some workaround instead of using `CaseIterable` if it's too much of a hassle.